File: SupplierPortal.Web/Modules/ServerTypes/Market/ItemForm.ts
import { IntegerEditor, StringEditor, PrefixedContext, initFormType } from "@serenity-is/corelib";

export interface ItemForm {
    ItemCode: IntegerEditor;
    ItemName: StringEditor;
}

export class ItemForm extends PrefixedContext {
    static readonly formKey = 'Market.Item';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!ItemForm.init)  {
            ItemForm.init = true;

            var w0 = IntegerEditor;
            var w1 = StringEditor;

            initFormType(ItemForm, [
                'ItemCode', w0,
                'ItemName', w1
            ]);
        }
    }
}

File: SupplierPortal.Web/Modules/Market/Item/ItemForm.cs
using Serenity.ComponentModel;

namespace SupplierPortal.Market.Forms;

[FormScript("Market.Item")]
[BasedOnRow(typeof(ItemRow), CheckNames = true)]
public class ItemForm
{
    public short ItemCode { get; set; }
    public string ItemName { get; set; }
}